projects/core/src/cart/event/active-cart-event.builder.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Type } from '@angular/core';
+import { filter, map } from 'rxjs/operators';
+import { EventService } from '../../event';
+import { ActiveCartService } from '../facade';
+import { ActiveCartAddEntryEvent } from './active-cart.events';
+import { MultiCartAddEntryEvent } from './multi-cart.events';
+
+/**
+ * Registers active cart events, when being injected
+ */
+@Injectable({ providedIn: 'root' })
+export class ActiveCartEventBuilder {
+  constructor(
+    protected event: EventService,
+    protected activeCartService: ActiveCartService
+  ) {
+    this.register();
+  }
+
+  /**
+   * Registers active cart events
+   */
+  protected register() {
+    return this.registerMapped(MultiCartAddEntryEvent, ActiveCartAddEntryEvent);
+  }
+
+  /**
+   * Registers a stream of target events mapped from the source events that contain the active cart id.
+   *
+   * @param sourceType type of the source event
+   * @param targetType type of the target event
+   */
+  private registerMapped<
+    S extends { cartId: string },
```

Comment:
Isn't the case that any cart event will have `cartId` string? Did you consider some base/generic cart event type?

Reply:
Created MultiCartEvent